docs/pages/database-access/guides/aws-dynamodb.mdx

```diff
@@ -0,0 +1,265 @@
+
+![Create Role to Identify EC2 Instance](../../../img/database-access/guides/dynamodb-create-ec2-role.png)
+
+This role requires no permissions, so on the next page you can click **Next**,
+then name the role and click **Create Role**. In this guide we'll use the example
+name `TeleportDatabaseService` for this role.
+
+### DynamoDB access role
+
+Navigate back to the Roles page and create a new role. Select the "AWS account"
+option, which creates a default trust policy to allow other entities in this
+account to assume this role:
+
+![Create Role Step 1](../../../img/database-access/guides/dynamodb-create-role-1.png)
+
+Press "Next". Find the AWS-managed policy `AmazonDynamoDBFullAccess` and then select the policy:
+
+![Create Role Step 2](../../../img/database-access/guides/dynamodb-create-role-2.png)
+
+<Admonition type="note" title="Apply least-privilege permissions">
+`AmazonDynamoDBFullAccess` may provide too much access for your intentions. To
+use a different IAM policy to reduce permissions, see [Managing access
+permissions to your Amazon DynamoDB
+Resources](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/access-control-overview.html)
+for more details.
+</Admonition>
+
+Press "Next". On the next page, enter a role name. In this guide we'll use
+the example name `ExampleTeleportDynamoDBRole` for this role.
+
+Under "Select trusted entities", update the JSON to allow the `TeleportDatabaseService`
+role to assume this role:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": [
+                    "arn:aws:iam::(=aws.aws_access_key=):role/ExampleTeleportDynamoDBRole"
+                ]
+            },
+            "Action": "sts:AssumeRole",
+            "Condition": {}
+        }
+    ]
+}
+```
+
+Finally, click "Create role".
+
+## Step 2/4. Configure the Teleport IAM role mapping
+
+The next step is to give your Teleport users permissions to assume IAM roles in
+your Teleport cluster.
+
+You can do this by creating a Teleport role with the `aws_role_arns` field
+listing the IAM role ARN created in the previous step. Create a file called
+`aws-dynamodb-access.yaml` with the following content:
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: aws-dynamodb-access
+spec:
+  allow:
+    app_labels:
+      '*': '*'
+    aws_role_arns:
+    - arn:aws:iam::(=aws.aws_access_key=):role/ExampleTeleportDynamoDBRole
+```
+
+<Details title="Templating aws_role_arns" opened={false}>
+The `aws_role_arns` field supports template variables so they can be populated
+dynamically based on your users' identity provider attributes. See [Role
+Templates](../../access-controls/guides/role-templates.mdx) for details.
+</Details>
+
+Create the new role:
+
+```code
+$ tctl create -f aws-dynamodb-access.yaml
+```
+
+(!docs/pages/includes/add-role-to-user.mdx role="aws-dynamodb-access"!)
```

Comment:
In other database access guides we typically have:
* Step 1: Install Teleport...
* Step 2: Create a Teleport User...
* Step 3: Create a database service configuration ...

and we just instruct them to grant the example user `db_users: '*'` in Teleport RBAC.

Additionally, this RBAC setup example is for application access. In Database access, you only need to allow the user to use a `db_user` that matches the name of the AWS IAM role for DynamoDB - in this example guide that would be `ExampleTeleportDynamoDBRole`. If you use the typical example where we grant `db_users: "*"` in Teleport RBAC, that will allow them to connect as any role for DynamoDB. I suggest sticking to that as it's likely they will choose a different AWS IAM role than the one in our examples.

I would suggest restructuring the guide like this:

```
## Step 1/X. Install Teleport

(!docs/pages/includes/database-access/start-auth-proxy.mdx!)

## Step 2/X. Create a Teleport user

(!docs/pages/includes/database-access/create-user.mdx!)

## Step 3/X. Install the Teleport Database Service
...

## Step 4/X. Create IAM roles for DynamoDB access
...
```

Reply:
I'm 👍 on restructuring as needed and updating the instructions for database access instead of application access norms. The one thing I'd push back on is the "Create a Teleport user" section. If the only reasoning is 'because we typically do it', I would push back on that. I think it's fair to assume that at this point in the reader's Teleport journey, they already have a Teleport user.  The prerequisites of this guide lay out the assumption that there is already a Teleport cluster in place, implying a user.

Reply:
Based on our slack discussion (and the issue/PR you are making as a a result of that), we can omit the additional setup steps. Just need to make sure we fix the RBAC steps to be db-access instead of app-access steps in this step now.

Reply:
I've updated the RBAC example as discussed. Additionally, I've expanded the Prerequisites section of this page to explicitly call for an existing Teleport cluster, with appropriate links. Please take a look, and mark this as resolved if you deem it so.